src/vivarium/framework/time.py

```diff
@@ -281,3 +286,31 @@ def simulant_next_event_times(self) -> Callable[[pd.Index], pd.Series]:
     def simulant_step_sizes(self) -> Callable[[pd.Index], pd.Series]:
         """Gets a callable that returns the simulant step sizes."""
         return self._manager.simulant_step_sizes
+
+    def register_step_modifier(
+        self,
+        modifier: Callable[[pd.Index], pd.Series],
+        requires_columns: List[str] = (),
+        requires_values: List[str] = (),
+        requires_streams: List[str] = (),
+    ) -> None:
+        """Registers a step size modifier.
+
+        Parameters
+        ----------
+        modifier
+            Modifier of the step size pipeline. Modifiers can take an index
+            and should return a series of step sizes.
+        requires_columns
+            A list of the state table columns that already need to be present
+            and populated in the state table before the modifier
+            is called.
+        requires_values
+            A list of the value pipelines that need to be properly sourced
+            before the  modifier is called.
+        requires_streams
+            A list of the randomness streams that need to be properly sourced
+            before the modifier is called."""
+        return self._manager.register_step_modifier(
```

Comment:
Just adding here what @hussain-jafari recommended: we should warn or maybe even error if the user tries to use `register_value_modifier("simulant_step_size", ...)` directly.

My opinion is to make a ticket for that since it seemed like we didn't have any great ideas on how to do that offhand.

Reply:
+ 1

Reply:
Filed at [MIC-4710](https://jira.ihme.washington.edu/browse/MIC-4710)